pkgs/servers/ankisyncd/default.nix

```diff
@@ -1,68 +1,58 @@
-{ lib
-, fetchFromGitHub
-, python3
-}:
+{ lib, stdenvNoCC, fetchFromGitHub, rustPlatform, protobuf }:
 
-python3.pkgs.buildPythonApplication rec {
+let
   pname = "ankisyncd";
-  version = "2.2.0";
-  src = fetchFromGitHub {
-    owner = "ankicommunity";
-    repo = "anki-sync-server";
-    rev = version;
-    hash = "sha256-RXrdJGJ+HMSpDGQBuVPPqsh3+uwAgE6f7ZJ0yFRMI8I=";
+  version = "1.1.3";
+
+  # anki-sync-server-rs expects anki sources in the 'anki' folder
+  # of its own source tree, with a patch applied (mostly to make
+  # some modules public): bundle our own src derivation manually
+  # We apparently cannot do that in postPatch or equivalent in a
+  # buildRustPackage derivation because the cargo deps are processed
+  # first, there might be a better way of doing this? But it works
+  ankiSrc = fetchFromGitHub {
+    owner = "ankitects";
+    repo = "anki";
+    rev = "2.1.60";
+    hash = "sha256-hNrf6asxF7r7QK2XO150yiRjyHAYKN8OFCFYX0SAiwA=";
     fetchSubmodules = true;
   };
```

Comment:
Could you add reasoning for why this is pinned rather than just taking `anki.src`?

Reply:
this has a few patches on top of anki, so needs a specific version to build -- if we update anki this will need to rebase the patches to build with the new version as well, and I'm not sure how timely upstream updates are in that respect (the sync protocol doesn't change all the time, so there should be no problem running with a slightly older anki version for its protocol definition, but I just haven't used this for long enough to tell)

I'll add a comment explaining this.